pallets/credit/src/lib.rs

```diff
@@ -1033,6 +1136,69 @@ pub mod pallet {
             Self::do_add_credit(account_id.clone(), credit_data);
             Self::deposit_event(Event::StakingCreditScore(account_id, credit));
         }
+
+        fn calc_usdt_daily_poc_reward(
+            account_id: &T::AccountId,
+            credit_data: &CreditData,
+        ) -> (BalanceOf<T>, Weight) {
+            let mut weight = Weight::zero();
+            let staking_balance = Self::user_staking_balance(account_id);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+            if staking_balance.is_none() {
+                return (0u32.into(), weight);
+            }
+            let staking_usdt = staking_balance.unwrap().0;
+            let price = Self::dpr_price();
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+            if price.is_none() {
+                return (0u32.into(), weight);
+            }
+            let dpr_amount =
+                staking_usdt / price.unwrap() * UniqueSaturatedFrom::unique_saturated_from(DPR);
+
+            let current_credit_level = credit_data.current_credit_level;
+            let credit_setting =
+                Self::credit_settings(credit_data.campaign_id, current_credit_level);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+
+            let daily_poc_reward = credit_setting.base_apy * dpr_amount / 365u32.into();
+            (daily_poc_reward, weight)
+        }
+
+        fn calc_normal_daily_poc_reward(credit_data: &CreditData) -> (BalanceOf<T>, Weight) {
+            let mut weight = Weight::zero();
+            let initial_credit_level = credit_data.initial_credit_level;
+            let credit_setting =
+                Self::credit_settings(credit_data.campaign_id, initial_credit_level);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+
+            // poc reward
+            let current_credit_level = credit_data.current_credit_level;
```

Comment:
Same like line 1159, no need to create a new variable.

Reply:
what't problem